Code/Framework/AzFramework/Platform/Linux/AzFramework/Windowing/NativeWindow_Linux_xcb.cpp

```diff
@@ -0,0 +1,244 @@
+                                                       s_XcbFormatDataSize, 
+                                                       1,
+                                                       &m_xcbAtomDeleteWindow);
+                                                    
+        AZ_Assert(ValidateXcbResult(xcb_check_result), "Failed to change atom property for WM_CLOSE event");
+
+        m_width = geometry.m_width;
+        m_height = geometry.m_height;
+    }
+
+    ////////////////////////////////////////////////////////////////////////////////////////////////
+    void NativeWindowImpl_Linux_xcb::Activate()
+    {
+        LinuxXcbEventHandlerBus::Handler::BusConnect();
+
+        if (!m_activated) // nothing to do if window was already activated
+        {
+            m_activated = true;
+
+            xcb_map_window(m_xcbConnection, m_xcbWindow);
+            xcb_flush(m_xcbConnection);
+        }
+    }
+
+    ////////////////////////////////////////////////////////////////////////////////////////////////
+    void NativeWindowImpl_Linux_xcb::Deactivate()
+    {
+        if (m_activated) // nothing to do if window was already deactivated
+        {
+            m_activated = false;
+
+            WindowNotificationBus::Event(reinterpret_cast<NativeWindowHandle>(m_xcbWindow), &WindowNotificationBus::Events::OnWindowClosed);
+
+            xcb_unmap_window(m_xcbConnection, m_xcbWindow);
+            xcb_flush(m_xcbConnection);
+        }
+        LinuxXcbEventHandlerBus::Handler::BusDisconnect();
+    }    
+
+    ////////////////////////////////////////////////////////////////////////////////////////////////
+    NativeWindowHandle NativeWindowImpl_Linux_xcb::GetWindowHandle() const
+    {
+        return reinterpret_cast<NativeWindowHandle>(m_xcbWindow);
+    }
+
+    ////////////////////////////////////////////////////////////////////////////////////////////////
+    void NativeWindowImpl_Linux_xcb::SetWindowTitle(const AZStd::string& title)
+    {
+        xcb_void_cookie_t xcb_check_result;
+        xcb_check_result = xcb_change_property(m_xcbConnection,
+                                               XCB_PROP_MODE_REPLACE,
+                                               m_xcbWindow,
+                                               XCB_ATOM_WM_NAME,
+                                               XCB_ATOM_STRING,
+                                               8,
+                                               static_cast<uint32_t>(title.size()),
+                                               title.c_str());
+        AZ_Assert(ValidateXcbResult(xcb_check_result), "Failed to set window title.");
+    }
+
+    ////////////////////////////////////////////////////////////////////////////////////////////////
+    void NativeWindowImpl_Linux_xcb::ResizeClientArea(WindowSize clientAreaSize)
+    {
+        const static uint32_t values[] = { clientAreaSize.m_width, clientAreaSize.m_height };
```

Comment:
Is the static necessary?

Reply:
not necessary, and probably not desirable since its being set by an argument. will update